src/store/workflows.module.js

```diff
@@ -386,11 +386,12 @@ function getFamilyTree (tokens, node) {
   }
 
   // add family levels below the cycle point
-  for (const ancestor of node.ancestors.slice().reverse()) {
+  for (let i = node.ancestors?.length ?? 0; i > 0; i--) {
+    const { name } = node.ancestors[i - 1]
     ret.push([
       'family',
-      ancestor.name,
-      lastTokens.clone({ task: ancestor.name })
+      name,
+      lastTokens.clone({ task: name })
     ])
   }
```

Comment:
Hmmm, this suggests that we are receiving an `added` delta for the family, in which `ancestors` is `undefined`? If so, we have a data store bug :(

Note, this is also relevant to the graph view which needs to request the family hierarchy.

By this logic:

* If `ancestors` is `undefined` on the first delta we receive, then we won't create the families under the cycle point.
* If `ancestors` is defined on a subsequent delta, that won't trigger this to subsequently run (because the node already exists in the store).

This will certainly give us a partial tree at first, but I'm not sure if something else in the store will subsequently force these nodes to be created on future deltas, so the issue might get fixed in time.

Reply:
(Discussed in person after some digging, summarising here for the record)

> Does this mean: The added delta just never got sent, something was lost in translation (family should be added to the store).

Yes, that. Possibly fixed by https://github.com/cylc/cylc-uiserver/pull/761

N.B. `updated` deltas for `root` families will never have `ancestors` as we have `updated(stripNull = true)` in the tree view subscription.

Btw, this PR will also defend against the bug if `stripNull` is ever turned on for `added` deltas